web-ts/src/routes/TuNewsPage.tsx

```diff
@@ -1,35 +1,152 @@
+  TU_NEWS_TYPE,
+  TunewsModel,
+  loadFromEndpoint
+} from 'api-client'
 import LocationLayout from '../components/LocationLayout'
-import LocationToolbar from '../components/LocationToolbar'
-import { FeedbackRatingType } from '../components/FeedbackToolbarItem'
+import DateFormatterContext from '../context/DateFormatterContext'
+import { useTranslation } from 'react-i18next'
+import NewsListItem from '../components/NewsListItem'
+import { createPath, TU_NEWS_DETAIL_ROUTE, TU_NEWS_ROUTE } from './index'
+import NewsTabs from '../components/NewsTabs'
+import { tunewsApiBaseUrl } from '../constants/urls'
+import LoadingSpinner from '../components/LoadingSpinner'
+import { FailureSwitcher } from '../components/FailureSwitcher'
+import TuNewsList from '../components/TuNewsList'
+
+const DEFAULT_PAGE = 1
+const DEFAULT_COUNT = 10
 
 type PropsType = {
   cities: Array<CityModel>
   cityModel: CityModel
   languages: Array<LanguageModel>
   languageModel: LanguageModel
-} & RouteComponentProps<{ cityCode: string; languageCode: string; newsId?: string }>
+} & RouteComponentProps<{ cityCode: string; languageCode: string }>
 
-const TuNewsPage = ({ match, cityModel, location }: PropsType) => {
-  const { languageCode } = match.params
-  const toolbar = (openFeedback: (rating: FeedbackRatingType) => void) => (
-    <LocationToolbar openFeedbackModal={openFeedback} viewportSmall={false} />
-  )
+const TuNewsPage = ({ match, cityModel, languages, location }: PropsType): ReactElement => {
+  const { cityCode, languageCode } = match.params
+  const pathname = normalizePath(location.pathname)
+  const formatter = useContext(DateFormatterContext)
+  const { t } = useTranslation('news')
+  const viewportSmall = false
+
+  const [tuNews, setTuNews] = useState<TunewsModel[]>([])
+  const [error, setError] = useState<Error | null>(null)
+  const [loading, setLoading] = useState<boolean>(false)
+  const [page, setPage] = useState<number>(DEFAULT_PAGE)
+  const [hasMore, setHasMore] = useState<boolean>(true)
+
+  const loadTuNews = useCallback(async () => {
+    if (hasMore) {
+      setLoading(true)
+      setPage(page + 1)
+      const endpoint = createTunewsEndpoint(tunewsApiBaseUrl)
+      const request = () => endpoint.request({ city: cityCode, language: languageCode, page, count: DEFAULT_COUNT })
+      const addTuNews = (data: TunewsModel[] | null) => {
+        if (data !== null) {
+          if (data.length === 0) {
+            setHasMore(false)
+          } else {
+            setTuNews(tuNews.concat(data))
+          }
+        }
+      }
+      await loadFromEndpoint(request, addTuNews, setError, setLoading)
+    }
+  }, [page, cityCode, languageCode, tuNews, setTuNews, setError, setLoading, hasMore])
+
+  useEffect(() => {
+    // Reset on language change
+    setTuNews([])
+    setError(null)
+    setLoading(false)
+    setHasMore(true)
+    setPage(DEFAULT_PAGE)
+  }, [languageCode, setTuNews, setError, setLoading, setHasMore, setPage])
```

Comment:
There is no reason th add the setters here and above, as the functions should always stay the same. You only need to add state, props and other actually changing variables to hook dependencies. Or just trust the "react-hooks/exhaustive-deps" eslint rule :)
```suggestion
  }, [languageCode])
```

Reply:
Hmmm... I was quite sure that the exhaustive-deps rule forced me to add these dependencies as well, but you are right, not needed...

Reply:
The documentation does not say that they do not change, so I'd just leave them: https://reactjs.org/docs/hooks-reference.html#usestate

Reply:
[From the docs](https://reactjs.org/docs/hooks-faq.html#what-can-i-do-if-my-effect-dependencies-change-too-often): 

> ```javascript
> function Counter() {
>   const [count, setCount] = useState(0);
> 
>   useEffect(() => {
>     const id = setInterval(() => {
>       setCount(c => c + 1); // ✅ This doesn't depend on `count` variable outside    }, 1000);
>     return () => clearInterval(id);
>   }, []); // ✅ Our effect doesn't use any variables in the component scope
>   return <h1>{count}</h1>;
> }
> ```
> (The identity of the setCount function is guaranteed to be stable so it’s safe to omit.)